internal/repository/employee/employee.go

```diff
@@ -1,53 +1,96 @@
 package employee
 
 import (
+	"context"
+	"errors"
 	"fmt"
 	"github.com/dilyara4949/employees-api/internal/domain"
-	"github.com/dilyara4949/employees-api/internal/repository/storage"
+	"sync"
 
 	"github.com/google/uuid"
 )
 
 type PositionsRepository interface {
-	Get(id string) (*domain.Position, error)
+	Get(ctx context.Context, id string) (*domain.Position, error)
 }
 
 type employeeRepository struct {
-	employeesStorage *storage.EmployeesStorage
+	employeesStorage *EmployeesStorage
 	positionsRepo    PositionsRepository
 }
 
-func NewEmployeesRepository(employeesStorage *storage.EmployeesStorage, positionsRepo PositionsRepository) domain.EmployeesRepository {
+func NewEmployeesRepository(employeesStorage *EmployeesStorage, positionsRepo PositionsRepository) domain.EmployeesRepository {
 	return &employeeRepository{employeesStorage: employeesStorage, positionsRepo: positionsRepo}
 }
 
-func (e *employeeRepository) Create(employee *domain.Employee) error {
-	if _, err := e.positionsRepo.Get(employee.PositionID); err != nil {
+type EmployeesStorage struct {
+	mu      sync.Mutex
+	storage map[string]domain.Employee
+}
+
+func NewEmployeesStorage() *EmployeesStorage {
+	return &EmployeesStorage{
+		storage: make(map[string]domain.Employee),
+	}
+}
+
+func (e *employeeRepository) Create(ctx context.Context, employee *domain.Employee) error {
+	if _, err := e.positionsRepo.Get(ctx, employee.PositionID); err != nil {
 		return fmt.Errorf("error to create employee: %w", err)
 	}
 
 	employee.ID = uuid.New().String()
-	e.employeesStorage.Add(*employee)
+	e.employeesStorage.mu.Lock()
```

Comment:
I think, we already discussed it, that you don't handle blocking and unblocking on the repo level, if you want to use storage.

Reply:
now I am confused more, after dividing locking/unlocking to separate folder, you said not to divide storage to separate folder since there is no database yet, i thought  that approach is what you asking to do. can you please explain why `repository/storage` wasn't correct, and how then it should be:

>  don't handle blocking and unblocking on the repo level

Reply:
fixed